man/systemd.special.xml

```diff
@@ -186,6 +186,13 @@
             encrypted block devices.</para>
           </listitem>
         </varlistentry>
+        <varlistentry>
+          <term><filename>veritysetup.target</filename></term>
+          <listitem>
+            <para>A target that pulls in setup services for all
+            verity block devices.</para>
+          </listitem>
+        </varlistentry>
```

Comment:
hmm, i wonder if we shouldn't just reuse cryptsetup.target for that. Do you have a strong reason to add a new target for this?

i.e. sure, it will be slightly weirdly named, but I'd keep things simple and say a single milestone for both verity and encrypted devices should suffice.

Reply:
Well fstab and crypttab have their own target (lots of other generators we supply do as well), so to me it would make sense to have one, so that it's clear they are related to different things.

Reply:
The drawback I see in creating a new target for `veritysetup` is the compatibility with what exists already.

The generated root units still reference to the previous target [cryptsetup-pre.target](https://github.com/systemd/systemd/blob/8374bf4fa2663b19f329801abfd8b29e66897380/src/veritysetup/veritysetup-generator.c#L100). Out of scope, maybe it should refer to `cryptsetup.target` directly?.

If we change the target it breaks the boot for people that uses initramfs because none of them include this newly introduced target `veritysetup.target`, and then, it creates a strong dependency between systemd and dracut for people using verity for root.

Anyway, I have some configuration files for dracut ready for that.

Reply:
i think distinct targets make sense if you want to handle what they group separately. I can see that for mounts and for crypt devices, since they are quite different, i.e. one is a thin kernel api, and the other requires a whole subsystem to work, device mgmt and everything. but the requirements for verity and cryptsetup are pretty much identical, i.e. is there any usecase where you#d want to manage the milestones for cryptsetup and veritysetups separately? I am not sure i see any...

I am not concerned about the dracut thing: we add new stuff all the time, dracut has to catch up, that's OK.

I am mostly interested about philosophically/conceptually if people ever want luks up but verity down or the opposite, or if they want verity up early and luks up late or so? i don't see that, do you?

Reply:
ah, well, let's leave this with distinct targets I guess. Maybe one day a usecase pops up, and it's better to have ti separate then already.

But please add counterparts for all of `cryptsetup-pre.target` + `cryptsetup.target` + `remote-cryptsetup.target`

Reply:
Speaking for my use case, I have a custom initramfs script that sets up verity and at some point in the near future I'd like to move to dracut + veritysetup generator. My disk grouped by cryptsetup.target is for external storage so to speak, not for the rootfs, and unlock is remote, so I could and in fact do have situations where my veritysetup targets need to be up (it's rootfs) and cryptsetup is not up yet or can't be up at all (first boot).